cellxgene_resources/cellxgene_mods.py

```diff
@@ -888,3 +897,61 @@ def evaluate_donors_sex(adata):
           )
 
         return donor_sex_df, dp
+
+
+def evaluate_var_df(adata):
+    """
+    Use single-cell-curation classes and fuctions and report warning/error for organism specific minimum number of gene features. Also, this function 
+    will look that var contains features from only a single organism.
+
+    :param obj adata: AnnData that is being curated
+
+    :return logging: Raises WARNING or ERROR if the number of genes in adata.var and adata.raw.var are fewer than the threshold of 40% or 60%, respectively,
+    of 10x preselected biotype of genes. Will also raise ERROR if dataset contains more than a singel organism.
+    """
+    accepted_biotypes = [
+        'protein_coding',
+        'protein_coding_LoF',
+        'lncRNA',
+        'IG_C_gene',
+        'IG_D_gene',
+        'IG_J_gene',
+        'IG_LV_gene',
+        'IG_V_gene',
+        'IG_V_pseudogene',
+        'IG_J_pseudogene',
+        'IG_C_pseudogene',
+        'TR_C_gene',
+        'TR_D_gene',
+        'TR_J_gene',
+        'TR_V_gene',
+        'TR_V_pseudogene',
+        'TR_J_pseudogene'
+    ]
+
+
+    # Check that this is single organism both in metadata and var index, exit function as does not make sense to check genes with multiple organisms
+    obs_organisms = adata.obs['organism_ontology_term_id'].unique()
+    var_organisms = {gencode.get_organism_from_feature_id(id).value for id in adata.var.index.to_list()}
+    if len(obs_organisms) > 1:
+        report(f'Multiple organisms found in obs metadata: {obs_organisms}', 'ERROR')
+        return
+    elif len(var_organisms) > 1:
+        report(f'Multiple organisms found in var index: {obs_organisms}', 'ERROR')
+        return
+    else:
+        report(f'Single organism found: {var_organisms}', 'GOOD')
+
+    # Check the number of genes threshold base on biotype per specific organism
+    organism = adata.obs['organism_ontology_term_id'].unique()[0]
+    org_obj = [i for i in gencode.SupportedOrganisms if i.value==organism][0]
+    gene_checker = gencode.GeneChecker(org_obj)
+    num_genes_biotype = len([i for i in gene_checker.gene_dict.keys() if gene_checker.gene_dict[i][2] in accepted_biotypes])
+
+    fraction = len(adata.var.index)/num_genes_biotype 
+    if fraction < 0.4:
+        report(f'only {len(adata.var.index)} out of {num_genes_biotype} 10x biotype genes, results in {fraction} (0.40 threshold)', 'ERROR')
+    elif fraction < 0.6:
+        report(f'{len(adata.var.index)} out of {num_genes_biotype} 10x biotype genes, results in {fraction} (0.60 threshold)','WARNING')
+    else:
+        report(f'{len(adata.var.index)} genes out of possible {num_genes_biotype} present', 'GOOD')
```

Comment:
I don't know if "possible" is the correct term here (Would be odd to see "60k genes out of possible 33k present")
Maybe "x genes present, compared against y genes"?
Maybe also display the fraction as that's what is being weighed?

Reply:
How about this:
`report(f'{len(adata.var.index)} genes present, compared against {num_genes_biotype} 10x biotype genes; fraction: {fraction}', 'GOOD')`